sklearn/linear_model/_ridge.py

```diff
@@ -235,6 +236,53 @@ def _solve_svd(X, y, alpha):
     return np.dot(Vt.T, d_UT_y).T
 
 
+def _solve_trf(
+    X, y, alpha, positive=False, max_iter=None, tol=1e-3, X_offset=None, X_scale=None
+):
+    lsq_config = {
+        "method": "trf",
+        "max_iter": max_iter,
+        "tol": tol,
+    }
+    n_samples, n_features = X.shape
+
+    if X_offset is None or X_scale is None:
+        X1 = sp_linalg.aslinearoperator(X)
+    else:
+        X1 = _get_rescaled_operator(X, X_offset, X_scale)
+
+    coefs = np.empty((y.shape[1], n_features), dtype=X.dtype)
+    Xa_shape = (n_samples + n_features, n_features)
+    if positive:
+        bounds = (0, np.inf)
+    else:
+        bounds = (-np.inf, np.inf)
+    sqrt_alpha = np.sqrt(alpha)
+
+    for i in range(y.shape[1]):
+
+        def mv(b):
+            return np.hstack([X1.matvec(b), sqrt_alpha[i] * b])
+
+        def rmv(b):
+            return X1.rmatvec(b[:n_samples]) + sqrt_alpha[i] * b[n_samples:]
+
+        Xa = sp_linalg.LinearOperator(shape=Xa_shape, matvec=mv, rmatvec=rmv)
+        y_zeros = np.zeros(n_features, dtype=X.dtype)
+        y_column = np.hstack([y[:, i], y_zeros])
+        result = sp_optimize.lsq_linear(Xa, y_column, bounds=bounds, **lsq_config)
+
+        if not result["success"]:
+            warnings.warn(
+                "trf solver did not converge. "
+                "Set a low `tol` and a high `max_iter`.",
```

Comment:
You mean a higher `tol`, right?

Also, it would be helpful to recall the actual values of tol and max_iter:

```suggestion
            warnings.warn(
                f"The trf solver did not converge. Try increasing max_iter "
                f"or tol. Currently: max_iter={max_iter} and tol={tol}"
```